chrome_extension/cardboard/popup.js

```diff
@@ -16,28 +43,40 @@ for (const tab of tabs) {
   if (tab.url != undefined) {
     element.querySelector(".puzzle_url").value = tab.url;
   }
+
+  // Add metas to dropdown
+  let meta_dropdown = element.querySelector(".puzzle_meta");
+  for (const meta of metas) {
+    let option = document.createElement("option");
+    option.text = meta;
+    option.value = meta;
+    option.key = meta;
+    meta_dropdown.add(option);
+  }
   elements.add(element);
 }
 document.querySelector("ul").append(...elements);
+
 const button = document.querySelector("button");
 button.addEventListener("click", async (e) => {
   e.preventDefault();
   // Get Cardboard cookie
   const cardboard_cookie = await chrome.cookies.get({
-    url: "https://cardboard.rocks",
+    url: URL,
     name: "csrftoken",
   });
 
   if (cardboard_cookie) {
     // Create puzzle. Puzzle name is limited to 30 characters
-    fetch("https://cardboard.rocks/api/v1/hunts/10/puzzles", {
+    fetch(URL + "/api/v1/hunts/" + HUNT_NUMBER + "/puzzles", {
       method: "POST",
       mode: "cors",
       body: JSON.stringify({
         create_channels: document.getElementById("create_channels").checked,
         is_meta: document.getElementById("is_meta").checked,
         name: document.getElementById("puzzle_name").value.slice(0, 30),
         url: document.getElementById("puzzle_url").value,
+        assigned_meta: document.getElementById("puzzle_meta").value,
       }),
       headers: {
         "X-CSRFToken": cardboard_cookie.value,
```

Comment:
it won't let me leave a review comment on the lines directly, but you should also delete the commented section at the bottom of this file

Reply:
Deleted